examples/routing_service/udp_socket_adapter/README.md

```diff
@@ -0,0 +1,165 @@
+[Visual Studio Solutions for Windows using CMake](https://cmake.org/cmake/help/v3.16/generator/Visual%20Studio%2016%202019.html#platform-selection).
+
+**Note:** `BUILD_SHARED_LIBS` allows you to control if the generated library
+for this example is a static or a dynamic shared library. The following
+sections assume you are building a dynamic shared library. However, Routing
+Service also supports static linking of adapters. To use this functionality
+you would need to create an application that uses Routing Service as a library
+component and statically links to this `SocketAdapter` library.
+
+### Cross-compilation
+
+When you need to cross-compile the example, the above
+command will not work, the assigned compiler won't be the cross-compiler and
+errors may happen when linking against the cross-compiled Connext binaries.
+To fix this, you have to create a file with the architecture name and call
+CMake with a specific flag called ``-DCMAKE_TOOLCHAIN_FILE``.
+An example of the file to create with the toolchain settings (e.g. for an
+ARM architectures):
+
+```cmake
+set(CMAKE_SYSTEM_NAME Linux)
+set(toolchain_path "<path to>/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian")
+set(CMAKE_C_COMPILER "${toolchain_path}/bin/arm-linux-gnueabihf-gcc")
+set(CMAKE_CXX_COMPILER "${toolchain_path}/bin/arm-linux-gnueabihf-g++")
+```
+
+Then you can call CMake like this:
+
+```bash
+cmake -DCONNEXTDDS_DIR=<connext dir> -DCMAKE_TOOLCHAIN_FILE=<toolchain file created above>
+      -DCONNEXTDDS_ARCH=<connext architecture> ..
+```
+
+## Running C++ example
+
+To run the example, you just need to run the following commands from the top
+level folder. This example has been written to allow easy experimentation with
+the Shapes Demo shipped with *RTI Connext DDS* installer bundle. You will find
+some hardcoded references to ShapeType and Square. If you wish to create a
+real Routing Service adapter, you should modify the code and XML accordingly.
+
+There is 1 configuration (`-cfgName`) in the Routing Service XML file:
+
+-   **SocketAdapterToDDS** - It reads data from a UDP socket using the
+SocketAdapter and outputs it to DDS. You can visualize the ouptut by
+subscribing to Squares in Shapes Demo or running:
+
+```bash
+ $NDDSHOME/bin/rtiddsspy -printSample
+```
+
+To run Routing Service, you will need first to set up your environment as
+follows.
+
+Before running the RTI Routing Service, you need to specify where the
+`SocketAdapterCpp` library is located as shown below:
+
+```bash
+$export RTI_LD_LIBRARY_PATH=$NDDSHOME/lib/<Connext DDS Architecture>:<Path to SocketAdapterCpp library>
+```
+
+The SocketAdapterCpp library will be in the `./build` folder.
+
+```bash
+# From the build/ directory
+$NDDSHOME/bin/rtiroutingservice -cfgFile RsSocketAdapter.xml -cfgName SocketAdapterToDDS
+```
+
+Here is an output from a sample run:
+
+```bash
+$export RTI_LD_LIBRARY_PATH=~/rti_connext_dds-7.3.0/lib/x64Linux4gcc7.3.0:~/udp_socket_adapter/build/
```

Comment:
This should make the instructions more future proof:

```suggestion
$ export RTI_LD_LIBRARY_PATH=$NDDSHOME/lib/$CONNEXT_ARCH:~/udp_socket_adapter/build/
```

Reply:
agreed!